static/src/javascripts/projects/common/modules/commercial/commercial-feature-policies.js

```diff
@@ -68,16 +68,15 @@ define([
         }
     };
 
-    policies.membershipMessages = function () {
-        if (!detect.adblockInUseSync() &&
-            detect.getBreakpoint() !== 'mobile' &&
-            config.page.contentType === 'Article' &&
-            !userFeatures.isPayingMember()
-        ) {
-            return {
-                membershipMessages : true
-            };
-        }
+    policies.checkWeCanShowMembershipMessages = function () {
```

Comment:
Perhaps `membershipMessages` should return `Promise[Boolean]` rather than return `Map[String, Promise[Boolean]]`? Unless you need to return a map/object, then the function could return `Promise[Map[String, Boolean]]`, so there is only one outer promise.

``` js
policies. checkWeCanShowMembershipMessages = function () {
    return detect.adblockInUse.then(adBlockInUse => {
        return !adBlockInUse
            && …
        }
    })
};
```